### PR TITLE
[Tests] Fixed usage of deprecated Curl::send & cleanup usage of PHPUnit5CompatTrait

### PR DESCRIPTION
**Pull request by MarioBlazek**

### CODE REVIEW

**Comment by andrerom** [comment]
Just for reference, but what introduced this error?

**Comment by MarioBlazek** [comment]
At first it looks like problem related to PHPUnit's convert notices to exceptions. Deprecated code was generating notices which resulted with error.

**Comment by MarioBlazek** [comment]
@alongosz please review.

File: eZ/Bundle/EzPublishRestBundle/Tests/Functional/TestCase.php [hunk 1]
```diff
@@ -88,7 +90,9 @@ protected function setUp()
     public function sendHttpRequest(HttpRequest $request)
     {
         $response = new HttpResponse();
-        $this->httpClient->send($request, $response);
+
+        $request = RequestConverter::psr7($request);
+        ResponseConverter::copy(ResponseConverter::buzz($this->httpClient->sendRequest($request, [])), $response);
```
Comment (alongosz) [inline eZ/Bundle/EzPublishRestBundle/Tests/Functional/TestCase.php]: Two issues I have doubts about:
1) Why do you need to copy `Response`? `::buzz` method should already give you correct object to return.
2) Second argument of `sendRequest` is optional and defaults to `[]` so it's not needed here.

I see the body of this method as:
```php
    public function sendHttpRequest(HttpRequest $request)
    {
        $request = RequestConverter::psr7($request);

        return ResponseConverter::buzz($this->httpClient->sendRequest($request));
    }
```

However, I'm not very familiar with Buzz, so maybe I'm missing something?

**Comment by MarioBlazek** [comment]
@alongosz PR updated as suggested.

**Comment by alongosz** [comment]
@andrerom @MarioBlazek It should also cause the same error on PRs backported to `7.0` (`master` is `7.1` now) because `phpunit-bridge` is required for `ezplatform:2.0`.

So I think this should be backported to `7.0`.

**Comment by MarioBlazek** [comment]
@alongosz should I change target branch to 7.0 ?

**Comment by alongosz** [comment]
Yep, and then rebase. I hope @andrerom agrees :)

**Comment by andrerom** [comment]
yes, I'm fine with that.

File: bootstrap.php [hunk 2]
```diff
@@ -20,4 +20,8 @@
     throw new \RuntimeException('Could not read config.php, please copy config.php-DEVELOPMENT to config.php & customize to your needs!');
 }
 
+// Class alias used for BC
+// // Enables old code which still extends non namespaced TestCase to work
+class_alias('PHPUnit\Framework\TestCase', 'PHPUnit_Framework_TestCase');
```
Comment (andrerom) [inline bootstrap.php]: hmm, where is this needed now? is it because of previous bump in phpunit version number?

Comment (MarioBlazek) [reply bootstrap.php]: @andrerom problem is with `AssetFactoryTest` it is extending from Symfony's `AssetFactoryTest` which extends from `PHPUnit_Framework_TestCase`.

Comment (andrerom) [reply bootstrap.php]: Ok, clear

**Pull request merged**